File: tests/test_data/multi_language/Calculator.swift
import Foundation

class Calculator {
    var result: Double = 0.0

    init() {
        result = 0.0
    }

    func add(_ value: Double) -> Calculator {
        result += value
        return self
    }

    static func create() -> Calculator {
        return Calculator()
    }
}

struct Point {
    var x: Double
    var y: Double
}

protocol MathOperations {
    func calculate(_ a: Double, _ b: Double) -> Double
}

enum Operation {
    case add, subtract, multiply, divide
}

extension Calculator: MathOperations {
    func calculate(_ a: Double, _ b: Double) -> Double {
        return a + b
    }
}

func standalone() -> Int {
    return 42
}

File: tests/test_data/multi_language/calculator.rb
# Ruby test file
module MathUtils
  class Calculator
    attr_reader :result

    def initialize
      @result = 0
    end

    def add(value)
      @result += value
      self
    end

    def multiply(value)
      @result *= value
      self
    end

    def self.create
      new
    end
  end
end
